lib/chunks.js

```diff
@@ -925,6 +925,27 @@ const ensureChunk = async (courseId, chunk) => {
     cwd: unpackPath,
   });
 
+  // Before we configure the symlink, we need to check if there are any
+  // outdated symlinks that need to be removed. Those can occur when a question
+  // is renamed into a directory nested inside of its former directory, e.g.
+  // if `questions/a/b/info.json` is moved to `questions/a/b/c/info.json`.
+  //
+  // We'll handle this by checking if any parent directory of the `targetPath`
+  // exists and is a symlink. If so, we'll remove it. This should always be
+  // safe because we should never have nested symlinks.
+  const pathSegments = relativeTargetPath.split(path.sep);
+  for (let i = 1; i < pathSegments.length; i++) {
+    const parentPath = path.join(courseChunksDirs.course, ...pathSegments.slice(0, i));
+    try {
+      const stat = await fs.lstat(parentPath);
+      if (stat.isSymbolicLink()) {
+        await fs.remove(parentPath);
+      }
```

Comment:
For an added check we could do something like:
```suggestion
      } else if (!stat.isDirectory()) {
        throw new Error(`${parentPath} exists but is not a directory`, stat);
      }
```

Reply:
Ah, interesting. That might actually be a scenario in which we should do something useful? Consider the following course, where `bar` is an extensionless file:

```
├── questions
│   ├── foo
│   │   ├── info.json
│   │   ├── bar
```

At some point that's changed to the following:

```
├── questions
│   ├── foo
│   │   ├── bar
│   │   │   ├── info.json
```

I'm pretty sure our code won't handle this case; maybe it should? This is admittedly a pretty contrived situation.

Reply:
Actually it turns out that this new code handles this just fine! I'll accept your suggestion then.